src/Symfony/Component/SignalRegistry/SignalRegistry.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\SignalRegistry;
+
+final class SignalRegistry implements SignalRegistryInterface
+{
+    private $signals = [];
+
+    public function __construct()
+    {
+        pcntl_async_signals(true);
+    }
+
+    public function register(int $signal, callable $callback): void
+    {
+        if (!isset($this->signals[$signal])) {
+            $previousCallback = pcntl_signal_get_handler($signal);
+            if (\is_callable($previousCallback)) {
+                $this->signals[$signal][] = $previousCallback;
+            }
+        }
+
+        $this->signals[$signal][] = $callback;
+        pcntl_signal($signal, [$this, 'handler']);
```

Comment:
`s/handler/handle/`

Reply:
I don't understand this comment, could you explain?

Reply:
@marie yes it means this :)
cf http://www.linfo.org/vi/search.html